finetrainers/trainer.py

```diff
@@ -741,14 +750,20 @@ def train(self) -> None:
                 transformer_lora_layers=transformer_lora_layers,
             )
 
+            self.validate(step=global_step, final_validation=True)
```

Comment:
I think this is it. Should not be called from inside `accelerator.is_main_process` otherwise `validate` will be deadlocked when the main process reaches its `accelerator.wait_for_everyone()` statement

Reply:
Oh I see. I thought this would be similar to how it's done in the `diffusers` script (i.e., validation run under the main process) but I forgot we have the luxury of distributed validation. 

Thanks for looking into this!